### PR TITLE
feat: RISCV native build support

### PR DESCRIPTION
**Pull request by SimonIT**
~not tested yet, it needs LWJGL/lwjgl3#890~

Everything's off to merge🎉

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -19,10 +19,11 @@ buildscript {
 	repositories {
 		mavenCentral()
 		gradlePluginPortal()
+		maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }
```
Comment (Tom-Ski) [inline build.gradle:22]: Can be removed

Comment (Tom-Ski) [reply build.gradle:22]: Ah we still need it for lwjgl3 release with that support. So probably can't be merged until then. Very cool though

Comment (SimonIT) [reply build.gradle]: lwjgl3 is released🎉

**Review by Berstanio — CHANGES_REQUESTED** [review]
jnigen 2.5.1 is broken with things like android package, we should wait merging this until 2.5.2 is released

**Comment by Tom-Ski** [comment]
@SimonIT we ready to merge this, jni gen 2.5.2 now pinned version for libgdx. Could you update this and we can test and get it in before relase

**Comment by SimonIT** [comment]
@Tom-Ski All conflicts are resolved!

File: CHANGES [hunk 2]
```diff
@@ -50,6 +50,7 @@
 - Android & iOS: Use empty Audio implementations on Android and iOS when audio is disabled
 - Adds status detection for an httpRequest
 - Fix ANGLE GLES renderer on dekstop (#7274)
+- Architecture support: Support for Linux RISC-V has been added. The gdx-xxx-natives.jar files now also contain native libraries for this architecture.
```
Comment (Berstanio) [inline CHANGES:53]: I would rather mention the gradle artifact here. That the `natives-desktop` now contains RISCV binaries.

**Review by Berstanio — APPROVED** [review]

**Review by Tom-Ski — APPROVED** [review]

**Comment by badlogic** [comment]
@SimonIT @Tom-Ski @Berstanio this is super cool! Sadly, the upgrade to LWJGL 3.3.4 breaks GLFW on macOS ARM64. The libglfw.dylibs shipped by LWJGL 3.3.4 lack some symbols, which will make things explode.

I.e. running `Lwjgl3TestStarter` on macOS ARM64 results in:

```
Caused by: java.lang.NullPointerException: A required function is missing: glfwGetWindowTitle
```

(Sorry for the screenshot, I can't get IDEA to print out the full stack trace in the console, or libGDX is swallowing it).
Screenshot 2024-08-20 at 11 37 12

Can we revert this until LWJGL has fixed their end? I've reported it on their Discord but haven't gotten a response yet.

**Comment by Berstanio** [comment]
Oh yeah, this is bad!
> Can we revert this until LWJGL has fixed their end?

This is all purely additional, so I think we can just revert, yes

**Comment by badlogic** [comment]
The plot thickens. A standalone LWJGL 3.3.4 app works. My guess: our libgdx
shared library loader doesn't unpack the LWJGL 3.3.4 libglfwarm64.dylib,
but instead reuses the 3.3.3 dylib that was already on my system from
previous runs. That'd be not so good.

**Pull request merged**